test/manifold/deferred_test.clj

```diff
@@ -239,6 +239,28 @@
     (is (= 1 @d))
     (is (= 1 (deref d 10 :foo)))))
 
+(deftest test-timeout
+  (testing "exception by default"
+    (let [d (d/deferred)
+          t (d/timeout! d 1)]
+      (is (identical? d t))
+      (is (thrown-with-msg? TimeoutException
+                            #"^timed out after 1 milliseconds$"
+                            @d))))
+
+  (testing "custom default value"
+    (let [d (d/deferred)
+          t (d/timeout! d 1 ::timeout)]
+      (is (identical? d t))
+      (is @(capture-success d ::timeout))))
+
+  (testing "error before timeout"
+    (let [ex (Exception.)
+          d (d/deferred)
+          t (d/timeout! d 1000)]
+      (d/error! d ex)
+      (is (= ex (deref (capture-error t) 100 ::error))))))
```

Comment:
do these numbers need to be so big?

if smaller works too that'd be better for faster tests

```suggestion
          t (d/timeout! d 20)]
      (d/error! d ex)
      (is (= ex (deref (capture-error t) 10 ::error))))))
```